services/microbadger/microbadger-size.tester.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const { isFileSize } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('image size without a specified tag')
+  .get('/fedora/apache.json')
+  .expectBadge({
+    label: 'image size',
+    message: isFileSize,
+  })
+
+t.create('image size with a specified tag')
+  .get('/fedora/apache/latest.json')
+  .expectBadge({
+    label: 'image size',
+    message: isFileSize,
+  })
+
+t.create('missing download size')
+  .get('/puppet/puppetserver.json')
+  .intercept(nock =>
+    nock('https://api.microbadger.com')
+      .get('/v1/images/puppet/puppetserver')
+      .reply(200, {})
+  )
+  .expectBadge({ label: 'image size', message: 'invalid response data' })
```

Comment:
I've switched this so we'll just fail validation in this case rather than having a special error case for it. Does this seem reasonable? Do we often see a case where the size is missing?

Reply:
It's quite a common case. If you're accessing an image (or the new version of an image) for the first time through Microbadger, the metadata will not be available and will be populated asynchronously. I believe it may take a few seconds for the information to be available through the API, you would basically need to query again after that period.

I think it would be ideal to have a special case saying "information not yet available" or something like that.

Reply:
Right - makes sense. Looking back over the code added https://github.com/badges/shields/pull/1536 it looks like this is a problem for size, but not layers and it will manifest by `DownloadSize` just not being present in the response at all (as opposed to `DownloadSize: null`), so I've tweaked the schema to handle that case.